SECURITY.md

```diff
@@ -0,0 +1,212 @@
+
+## Scope and Supported Versions
+
+We (the community) support the 4 most recent minor versions of Knative. For more
+details on what this means, see our
+[release principles](https://knative.dev/community/contributing/mechanics/release-versioning-principles/).
+
+While we encourage you to report vulnerabilities in any Knative repo, at HEAD or
+the tip of any release branch (release-X.Y), we commit to patching
+vulnerabilities only at HEAD and in currently supported release branches.
+
+Repos with releases have the releases listed in the Github UI, e.g.:
+
+- [serving](https://github.com/knative/serving/releases)
+- [eventing](https://github.com/knative/eventing/releases)
+
+Repos in Knative Sandbox that have no releases listed may have a higher bar for
+accepting vulnerability reports. No support is provided for archived repos.
+
+## Reporting a Vulnerability
+
+Please email the
+[Knative Security list](mailto:knative-security@googlegroups.com) with the
+following information:
+
+- the repo and version/branch that contains the vulnerability
+- a description of the type of vulnerability (e.g., buffer overflow)
+
+_Please avoid including any details which would allow reproduction of the issue
+at this stage._ Details will be requested subsequently.
+
+Someone from the list will get back to you within 3 business days. Accepted
+reports will be fixed within 90 calendar days via one or more releases for
+versions that are both affected and supported at that point.
+
+We ask that you keep reports confidential for the full 90 days, or until the
+security list members release you from the confidentiality, whichever comes
+first, as we coordinate disclosures with Knative partners who may need some time
+after the vulnerability is addressed in Knative in order to update their
+systems.
+
+### What's in a report?
+
+Initial reports via email should include the limited information described
+[above](#reporting-a-vulnerability). Followup details should include a
+description of the vulnerability, a proof-of-concept exploit demonstrating the
+vulnerability, and a statement of what this vulnerability puts at risk. Please
+send reports in English.
+
+The security list members may make a best-effort attempt to address reports that
+do not meet these standards.
+
+### Coordinated Vulnerability Disclosure
+
+Knative Security engages with partners, including some vendors of
+[Knative-based offerings](https://knative.dev/docs/knative-offerings/), to
+responsibly disclose newly discovered vulnerabilities.
+
+If you believe your organization should participate in Knative's Coordinated
+Vulnerability Disclosure (CVD) process, please contact Knative Security to
+discuss this.
+
+#### Dependencies
+
+From time to time, Knative dependencies may have their own vulnerabilities.
+Those coordinating the response to and disclosure of those vulnerabilities are
+encouraged to reach out to Knative Security through our
```

Comment:
I guess related to the above question, how will our deps know that they are our dependency. Should there be a way for us to register a dependency? I have seen a dependabot file an issue (and fixed it), is that the way these things are best kept track of? Or are there other best practices.

Reply:
I don't really know yet, but it's clearly something we need to do, and needs to be on the Security WG agenda to figure out.

`go mod` support is coming to dependabot, but isn't here yet. IIUC there are probably dependabot alternatives until then. The CodeQL & lgtm.com stuff we turned on last summer may also find some issues.

https://lgtm.com/projects/g/knative/serving/ 
https://lgtm.com/projects/g/knative/serving/ci/

https://docs.github.com/en/github/managing-security-vulnerabilities/configuring-dependabot-security-updates
https://docs.github.com/en/github/visualizing-repository-data-with-graphs/about-the-dependency-graph#supported-package-ecosystems
https://dependabot.com/go/

Reply:
Anyhow this is a question for the WG, and not something to be answered in this policy doc.